torch_geometric/distributed/dist_loader.py

```diff
@@ -0,0 +1,103 @@
+import atexit
+import os
+from typing import Dict, List, Optional, Union
+
+import torch
+import torch.multiprocessing as mp
+
+from .dist_context import DistContext, DistRole
+from .dist_neighbor_sampler import DistNeighborSampler, close_sampler
+from .rpc import global_barrier, init_rpc
+
+
+class DistLoader():
+    def __init__(
+            self,
+            neighbor_sampler: DistNeighborSampler,
+            current_ctx: DistContext,
+            rpc_worker_names: Dict[DistRole, List[str]],
+            master_addr: str,
+            master_port: Union[int, str],
+            # Optional[Union[ChannelBase, mp.Queue()]],
+            channel: mp.Queue(),
+            num_rpc_threads: Optional[int] = 16,
+            rpc_timeout: Optional[int] = 180,
+            device: Optional[torch.device] = torch.device('cpu'),
+            **kwargs,
+    ):
+
+        self.neighbor_sampler = neighbor_sampler
+        self.channel = channel
+
+        self.current_ctx = current_ctx
+        self.rpc_worker_names = rpc_worker_names
+        self.pid = mp.current_process().pid
+        self.device = device
+        self.batch_size = kwargs.get('batch_size', 64)
+        self.num_workers = kwargs.get('num_workers', 0)
```

Comment:
Wouldn't it make sense to inherit the default values from `torch.utils.data.DataLoader`? They also don't seem to be used anywhere, so why do need them?

Reply:
I removed the `batch_size`, as it wasn't used, but the `num_workers` is required at initialization (L90-92) for the case when `num_workers`==0 and this condition is evaluated before `DistLoader` init.